packages/core/src/ActionsGeneric/ActionsGeneric.js

```diff
@@ -28,15 +29,17 @@ const ActionsGeneric = ({
     return (
       <HvButton
         id={actionId}
+        icon={iconButtons}
         key={actionId || idx}
         category={category}
         className={classes.button}
         disabled={actDisabled ?? disabled}
         onClick={(event) => actionsCallback?.(event, id, action)}
-        startIcon={renderedIcon}
+        startIcon={iconButtons ? undefined : renderedIcon}
+        aria-label={iconButtons ? label : undefined}
         {...other}
       >
-        {label}
+        {iconButtons ? renderedIcon : label}
```

Comment:
To improve this, we could use the label as a condition to understand whether the user wants an icon button or a button with text and icon, by switching between` label == null ? renderedIcon : label`, with this we can avoid increasing a new prop to the API check: https://github.com/lumada-design/hv-uikit-react/commit/e2497590742d7cac812e2d4314c5f49357299b1a